src/services/selfHosted/useSelfHostedSeatsConfig.ts

```diff
@@ -0,0 +1,54 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+export const SeatsSchema = z
+  .object({
+    config: z
+      .object({
+        seatsUsed: z.number().nullable(),
+        seatsLimit: z.number().nullable(),
+      })
+      .nullable(),
+  })
+  .nullable()
+
+const query = `
+query Seats {
+  config {
+    seatsUsed
+    seatsLimit
+  }
+}
+`
+
+interface URLParams {
+  provider: string
+}
+
+export const useSelfHostedSeatsConfig = (options = {}) => {
+  const { provider } = useParams<URLParams>()
+
+  return useQuery({
+    queryKey: ['Seats', provider, query],
+    queryFn: ({ signal }) =>
+      Api.graphql({ provider, query, signal }).then((res) => {
+        const parsedRes = SeatsSchema.safeParse(res?.data)
+
+        if (!parsedRes.success) {
+          return Promise.reject({
+            status: 404,
+            data: {},
+            dev: 'useSelfHostedSeatsConfig - 404 schema parsing failed',
+          } satisfies NetworkErrorObject)
+        }
+
+        return parsedRes ?? null
```

Comment:
I think we need to use the `parsedRes.data` field here so we only get the data sent from the API over the that plus the zod info

Reply:
`select` below it was pulling out `data`. I adjusted these 2 to fit the pattern in the rest of our app